examples/fluids/qfunctions/stabilization.h

```diff
@@ -0,0 +1,167 @@
+// *****************************************************************************
+CEED_QFUNCTION_HELPER void Tau_diagPrim(NewtonianIdealGasContext gas, State s,
+                                        const CeedScalar dXdx[3][3],
+                                        const CeedScalar dt, CeedScalar Tau_d[3]) {
+  // Context
+  const CeedScalar Ctau_t = gas->Ctau_t;
+  const CeedScalar Ctau_v = gas->Ctau_v;
+  const CeedScalar Ctau_C = gas->Ctau_C;
+  const CeedScalar Ctau_M = gas->Ctau_M;
+  const CeedScalar Ctau_E = gas->Ctau_E;
+  const CeedScalar cv = gas->cv;
+  const CeedScalar mu = gas->mu;
+  const CeedScalar u[3] = {s.Y.velocity[0], s.Y.velocity[1], s.Y.velocity[2]};
+  const CeedScalar rho = s.U.density;
+
+  CeedScalar gijd[6];
+  CeedScalar tau;
+  CeedScalar dts;
+  CeedScalar fact;
+
+  //*INDENT-OFF*
+  gijd[0] =   dXdx[0][0] * dXdx[0][0]
+            + dXdx[1][0] * dXdx[1][0]
+            + dXdx[2][0] * dXdx[2][0];
+
+  gijd[1] =   dXdx[0][0] * dXdx[0][1]
+            + dXdx[1][0] * dXdx[1][1]
+            + dXdx[2][0] * dXdx[2][1];
+
+  gijd[2] =   dXdx[0][1] * dXdx[0][1]
+            + dXdx[1][1] * dXdx[1][1]
+            + dXdx[2][1] * dXdx[2][1];
+
+  gijd[3] =   dXdx[0][0] * dXdx[0][2]
+            + dXdx[1][0] * dXdx[1][2]
+            + dXdx[2][0] * dXdx[2][2];
+
+  gijd[4] =   dXdx[0][1] * dXdx[0][2]
+            + dXdx[1][1] * dXdx[1][2]
+            + dXdx[2][1] * dXdx[2][2];
+
+  gijd[5] =   dXdx[0][2] * dXdx[0][2]
+            + dXdx[1][2] * dXdx[1][2]
+            + dXdx[2][2] * dXdx[2][2];
+  //*INDENT-ON*
+
+  dts = Ctau_t / dt ;
+
+  tau = rho*rho*((4. * dts * dts)
+                 + u[0] * ( u[0] * gijd[0] + 2. * ( u[1] * gijd[1] + u[2] * gijd[3]))
+                 + u[1] * ( u[1] * gijd[2] + 2. *   u[2] * gijd[4])
+                 + u[2] *   u[2] * gijd[5])
+        + Ctau_v* mu * mu *
+        (gijd[0]*gijd[0] + gijd[2]*gijd[2] + gijd[5]*gijd[5] +
+         + 2. * (gijd[1]*gijd[1] + gijd[3]*gijd[3] + gijd[4]*gijd[4]));
+
+  fact = sqrt(tau);
+
+  Tau_d[0] = Ctau_C * fact / (rho*(gijd[0] + gijd[2] + gijd[5]))*0.125;
+
+  Tau_d[1] = Ctau_M / fact;
+  Tau_d[2] = Ctau_E / ( fact * cv );
+
+  // consider putting back the way I initially had it  Ctau_E * Tau_d[1] /cv
+  //  to avoid a division if the compiler is smart enough to see that cv IS
+  // a constant that it could invert once for all elements
+  // but in that case energy tau is scaled by the product of Ctau_E * Ctau_M
+  // OR we could absorb cv into Ctau_E but this puts more burden on user to
+  // know how to change constants with a change of fluid or units.  Same for
+  // Ctau_v * mu * mu IF AND ONLY IF we don't add viscosity law =f(T)
+}
+
+// *****************************************************************************
+// Helper function for computing flux Jacobian of Primitive variables
+// *****************************************************************************
+CEED_QFUNCTION_HELPER void computeFluxJacobian_NSp(CeedScalar dF[3][5][5],
```

Comment:
Do we need to keep this?

Reply:
It doesn't get called. We might want such a thing when implementing an eigenvalue/matrix-sqrt stabilization operator, though we can resurrect it from the history if we don't have a better way to compute the matrix at that time.